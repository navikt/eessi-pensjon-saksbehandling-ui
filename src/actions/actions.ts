export interface Action<T> {
  type: string,
  payload?: T
}

export interface JustTypeAction {
  type: string
}

export interface PayloadError {
  error: any
}
