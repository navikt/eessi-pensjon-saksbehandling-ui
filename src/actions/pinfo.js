import * as types from 'constants/actionTypes'
import * as urls from 'constants/urls'
import * as api from 'eessi-pensjon-ui/dist/api'
const sprintf = require('sprintf-js').sprintf

export const sendInvite = (params) => {
  return api.call({
    url: sprintf(urls.API_VARSEL_URL, params),
    method: 'POST',
    payload: {},
    expectedPayload: { success: true },
    type: {
      request: types.PINFO_INVITE_REQUEST,
      success: types.PINFO_INVITE_SUCCESS,
      failure: types.PINFO_INVITE_FAILURE
    }
  })
}
