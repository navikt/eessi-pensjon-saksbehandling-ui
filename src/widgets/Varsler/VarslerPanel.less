.w-varslerPanel {
  &__title {
    display: flex;
    justify-content: space-between;
  }
}
