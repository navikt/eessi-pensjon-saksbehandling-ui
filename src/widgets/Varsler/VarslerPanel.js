import React, { useEffect, useState } from 'react'
import PT from 'prop-types'
import { connect, bindActionCreators } from 'store'
import _ from 'lodash'
import moment from 'moment'
import { withTranslation } from 'react-i18next'
import { Icons, Nav, RefreshButton, WaitingPanel } from 'eessi-pensjon-ui'
import * as pinfoActions from 'actions/pinfo'
import * as storageActions from 'actions/storage'
import { ReactComponent as VeilederSVG } from 'resources/images/NavPensjonVeileder.svg'
import './VarslerPanel.css'

const mapStateToProps = /* istanbul ignore next */ (state) => {
  return {
    aktoerId: state.app.params.aktoerId,
    fileList: state.storage.fileList,
    file: state.storage.file,
    invite: state.pinfo.invite,
    isInvitingPinfo: state.loading.isInvitingPinfo,
    sakId: state.app.params.sakId,
    sakType: state.app.params.sakType,
    person: state.app.person
  }
}

const mapDispatchToProps = /* istanbul ignore next */ (dispatch) => {
  return { actions: bindActionCreators({ ...pinfoActions, ...storageActions }, dispatch) }
}

export const VarslerPanel = (props) => {
  const { actions, aktoerId, file, fileList, isInvitingPinfo, invite, onUpdate, person, sakId, sakType, t, widget } = props

  const [isReady, setIsReady] = useState(false)
  const [_fileList, setFileList] = useState(undefined)
  const [_files, setFiles] = useState({})
  const [mounted, setMounted] = useState(false)
  const [hasParams, setHasParams] = useState(false)

  const onInviteButtonClick = () => {
    actions.sendInvite({
      aktoerId: aktoerId,
      sakId: sakId
    })
  }

  const onExpandablePanelChange = () => {
    const newWidget = _.cloneDeep(widget)
    newWidget.options.collapsed = !newWidget.options.collapsed
    onUpdate(newWidget)
  }

  useEffect(() => {
    if (!mounted) {
      if (aktoerId && sakId && fileList === undefined) {
        actions.listStorageFiles({
          userId: aktoerId,
          namespace: 'varsler___' + sakId
        })
      }
      setHasParams(aktoerId !== undefined && sakId !== undefined)
      setMounted(true)
    }
  }, [actions, aktoerId, sakId, fileList, mounted])

  useEffect(() => {
    if (fileList !== undefined && _fileList === undefined) {
      fileList.map(file => {
        actions.getStorageFile({
          userId: aktoerId,
          namespace: 'varsler',
          file: sakId + '___' + file
        }, {
          notification: false
        })
        return file
      })

      setIsReady(_.isEmpty(fileList))
      setFileList(fileList)
    }

    if (file !== undefined && !isReady) {
      const files = _.cloneDeep(_files)
      const key = file.timestamp + '.json'
      if (!Object.prototype.hasOwnProperty.call(files, key)) {
        files[key] = file
        const allFilesDone = Object.keys(files).length === fileList.length
        setFiles(files)
        setIsReady(allFilesDone)
      }
    }
  }, [actions, aktoerId, fileList, _fileList, _files, file, isReady, sakId])

  const onRefreshHandle = () => {
    if (aktoerId && sakId) {
      setFileList(undefined)
      setFiles({})
      setIsReady(false)
      actions.listStorageFiles({
        userId: aktoerId,
        namespace: 'varsler___' + sakId
      })
    }
  }

  if (!mounted) {
    return null
  }

  if (mounted && !hasParams) {
    return (
      <Nav.Panel className='w-varslerPanel s-border'>
        <Nav.Systemtittel className='w-varslerPanel__noParams-title pb-3'>{t('ui:widget-overview-notifications')}</Nav.Systemtittel>
        <Nav.Veileder
          tekst={t('pinfo:error-noParams')}
          posisjon='hÃ¸yre'
          type='feilmelding'
          fargetema='feilmelding'
        >
          <VeilederSVG />
        </Nav.Veileder>
      </Nav.Panel>
    )
  }

  const user = _.get(person, 'aktoer.ident.ident') || '-'
  return (
    <Nav.EkspanderbartpanelBase
      className='w-varslerPanel s-border'
      apen={!widget.options.collapsed}
      onClick={onExpandablePanelChange}
      heading={<Nav.Systemtittel className='pb-3'>{t('ui:widget-overview-notifications')}</Nav.Systemtittel>}
    >
      <Nav.Row>
        <div className='col-md-4'>
          <div style={{ display: 'none' }}>
            <div><label className='skjemaelement__label d-inline-block'>{t('pinfo:sb-sakId')}</label>: {sakId}</div>
            <div><label className='skjemaelement__label d-inline-block'>{t('ui:aktoerId')}</label>: {aktoerId}</div>
            <div><label className='skjemaelement__label d-inline-block'>{t('pinfo:sb-sakType')}</label>: {sakType}</div>
          </div>
          <Nav.Veileder
            tekst={(
              <>
                {!_.isEmpty(invite)
                  ? <div dangerouslySetInnerHTML={{ __html: t(invite.message) }} />
                  : <div dangerouslySetInnerHTML={{ __html: t('ui:widget-overview-sendNotification-description', { user: user }) }} />}
              </>
            )}
            type={invite ? (invite.status === 'ERROR' ? 'feilmelding' : 'suksess') : 'normal'}
            fargetema={invite ? (invite.status === 'ERROR' ? 'feilmelding' : 'suksess') : 'normal'}
            posisjon='bunn'
          >
            <VeilederSVG />
          </Nav.Veileder>
          <div className='text-center'>
            <Nav.Hovedknapp
              id='w-varslerPanel__invite-button-id'
              className='w-varslerPanel__invite-button mt-3'
              disabled={isInvitingPinfo}
              spinner={isInvitingPinfo}
              onClick={onInviteButtonClick}
            >
              {isInvitingPinfo ? t('sending') : t('ui:widget-overview-sendNotification-button')}
            </Nav.Hovedknapp>
          </div>
        </div>
        <div className='col-md-8'>
          <div className='w-varslerPanel__title'>
            {/* <Undertittel>{t('ui:widget-overview-sendNotification-title')}</Undertittel> */}
            <div />
            <RefreshButton
              className='w-varslerPanel__refresh-button'
              labelRefresh={t('ui:refresh')}
              rotating={!isReady}
              onRefreshClicked={onRefreshHandle}
            />
          </div>
          {!isReady ? (
            <WaitingPanel style={{ paddingTop: '3rem' }} message={t('ui:loading')} />
          ) : (
            <table className='w-100 mt-2'>
              <thead>
                <tr style={{ borderBottom: '1px solid lightgrey' }}>
                  <th />
                  <th>{t('ui:sentDocuments')}</th>
                  <th>{t('ui:sender')}</th>
                  <th>{t('ui:dateandtime')}</th>
                </tr>
              </thead>
              <tbody>
                {_files && !_.isEmpty(_files)
                  ? (
                    Object.keys(_files)
                      .sort((a, b) => _files[b].timestamp.localeCompare(_files[a].timestamp))
                      .map((file, index) => {
                        const content = _files[file]
                        return (
                          <tr className='slideAnimate' style={{ animationDelay: index * 0.03 + 's' }} key={file}>
                            <td><Icons kind='nav-message-sent' /></td>
                            <td>{content.tittel || file}</td>
                            <td>{content.fulltnavn || t('unknown')}</td>
                            <td>{content.timestamp ? moment(content.timestamp).format('DD.MM.YYYY') : t('unknown')}</td>
                          </tr>
                        )
                      })
                  ) : (
                    <tr>
                      <td colSpan={4} align='center' className='p-2 font-italic'>{t('ui:widget-overview-sendNotification-noMessages')}</td>
                    </tr>
                  )}
              </tbody>
            </table>
          )}
        </div>
      </Nav.Row>
    </Nav.EkspanderbartpanelBase>
  )
}

VarslerPanel.propTypes = {
  actions: PT.object.isRequired,
  aktoerId: PT.string,
  file: PT.object,
  fileList: PT.array,
  isInvitingPinfo: PT.bool,
  invite: PT.object,
  person: PT.object,
  sakId: PT.string,
  sakType: PT.string,
  t: PT.func.isRequired,
  widget: PT.object.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(withTranslation()(VarslerPanel))
