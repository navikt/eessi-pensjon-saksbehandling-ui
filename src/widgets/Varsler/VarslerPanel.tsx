import { sendInvite } from 'actions/pinfo'
import { getStorageFile, listStorageFiles } from 'actions/storage'
import LineMessageSent from 'assets/icons/line-version-expanded-email-send-3'
import ExpandingPanel from 'components/ExpandingPanel/ExpandingPanel'
import RefreshButton from 'components/RefreshButton/RefreshButton'
import { HighContrastHovedknapp, HorizontalSeparatorDiv, VerticalSeparatorDiv } from 'components/StyledComponents'
import { Widget } from 'nav-dashboard'
import _ from 'lodash'
import moment from 'moment'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'
import { State } from 'declarations/reducers'
import { ReactComponent as VeilederSVG } from 'assets/images/NavPensjonVeileder.svg'
import styled from 'styled-components'
import TableSorter, { Item } from 'tabell'
import Panel from 'nav-frontend-paneler'
import { Systemtittel } from 'nav-frontend-typografi'
import Veileder from 'nav-frontend-veileder'

const mapState = /* istanbul ignore next */ (state: State) => ({
  aktoerId: state.app.params.aktoerId,
  fileList: state.storage.fileList,
  file: state.storage.file,
  invite: state.pinfo.invite,
  isInvitingPinfo: state.loading.isInvitingPinfo,
  sakId: state.app.params.sakId,
  sakType: state.app.params.sakType,
  person: state.app.person
})

export interface Varsler {
  tittel: string
  fulltnavn: string
  timestamp: string
}

export type Varslers = {[k: string]: Varsler}

export const VarslerPropType = PT.shape({
  tittel: PT.string.isRequired,
  fulltnavn: PT.string.isRequired,
  timestamp: PT.string.isRequired
})

interface Invite {
  status: string
  message: string
}

export const InvitePropType = PT.shape({
  status: PT.string.isRequired,
  message: PT.string.isRequired
})

export interface VarslerPanelSelector {
  aktoerId: string
  fileList: Array<string> | undefined
  file: any
  invite: Invite | undefined
  isInvitingPinfo: boolean
  sakId: string
  sakType: string | undefined
  person: any
}

export interface VarslerPanelProps {
  highContrast: boolean
  initialFiles?: Varslers
  onUpdate?: (w: Widget) => void
  widget: Widget
}

const WidgetPanel = styled(Panel)`
  border: 1px solid ${({ theme }: any) => theme.navGra60};
`
const Title = styled.div`
  display: flex;
  justify-content: space-between;
`
const Table = styled(TableSorter)`
  width: 100%;
  margin-top: 0,.5rem;
  tr, td {
    padding: 0.5rem;
  }
`
const NoMessage = styled.div`
  padding: 0.5rem;
  font-style: italic;
  text-align: center;
`
const FlexDiv = styled.div`
  display: flex;
`

export const VarslerPanel: React.FC<VarslerPanelProps> = ({
  highContrast, initialFiles = {}, onUpdate, widget
}: VarslerPanelProps) => {
  const { aktoerId, file, fileList, isInvitingPinfo, invite, person, sakId, sakType }: VarslerPanelSelector = useSelector<State, VarslerPanelSelector>(mapState)
  const [isReady, setIsReady] = useState(false)
  const [_fileList, setFileList] = useState<Array<string> | undefined | null>(undefined)
  const [_files, setFiles] = useState<Varslers>(initialFiles)
  const [mounted, setMounted] = useState(false)
  const [hasParams, setHasParams] = useState(false)

  const { t } = useTranslation()
  const dispatch = useDispatch()
  const onInviteButtonClick = () => {
    dispatch(sendInvite({
      aktoerId: aktoerId,
      sakId: sakId
    }))
  }

  const onExpandablePanelChange = () => {
    const newWidget = _.cloneDeep(widget)
    newWidget.options.collapsed = !newWidget.options.collapsed
    if (onUpdate) {
      onUpdate(newWidget)
    }
  }

  useEffect(() => {
    if (!mounted) {
      if (aktoerId && sakId && fileList === undefined) {
        dispatch(listStorageFiles({
          userId: aktoerId,
          namespace: 'varsler___' + sakId
        }))
      }
      setHasParams(aktoerId !== undefined && sakId !== undefined)
      setMounted(true)
    }
  }, [aktoerId, dispatch, sakId, fileList, mounted])

  useEffect(() => {
    if (!_.isNil(fileList) && _fileList === undefined) {
      fileList.forEach(file => {
        if (!_.includes(Object.keys(_files), file)) {
          dispatch(getStorageFile({
            userId: aktoerId,
            namespace: 'varsler',
            file: sakId + '___' + file
          }, {
            notification: false
          }))
        }
      })

      setIsReady(_.isEmpty(fileList))
      setFileList(fileList)
    }

    if (!_.isNil(fileList) && file !== undefined && !isReady) {
      const files = _.cloneDeep(_files)
      const key = file.timestamp + '.json'
      if (!Object.prototype.hasOwnProperty.call(files, key)) {
        files[key] = file
        const allFilesDone = Object.keys(files).length === fileList.length
        setFiles(files)
        setIsReady(allFilesDone)
      }
    }
  }, [aktoerId, dispatch, fileList, _fileList, _files, file, isReady, sakId])

  const onRefreshHandle = () => {
    if (aktoerId && sakId) {
      setFileList(undefined)
      setFiles({})
      setIsReady(false)
      dispatch(listStorageFiles({
        userId: aktoerId,
        namespace: 'varsler___' + sakId
      }))
    }
  }

  if (!mounted) {
    return null
  }

  if (mounted && !hasParams) {
    return (
      <WidgetPanel>
        <Systemtittel className='w-varslerPanel__noParams-title pb-3'>{t('ui:widget-overview-notifications')}</Systemtittel>
        <Veileder
          tekst={t('pinfo:error-noParams')}
          posisjon='hÃ¸yre'
          type='feilmelding'
          fargetema='feilmelding'
        >
          <VeilederSVG />
        </Veileder>
      </WidgetPanel>
    )
  }

  const user = _.get(person, 'aktoer.ident.ident') || '-'

  const items = _files && !_.isEmpty(_files) ? Object.keys(_files)
    .sort((a, b) => _files[b].timestamp.localeCompare(_files[a].timestamp))
    .map((file) => {
      const content = _files[file]
      return {
        key: file,
        type: content.tittel || file,
        sender: content.fulltnavn || t('unknown'),
        date: moment(content.timestamp).toDate()
      }
    }) : []

  return (
    <ExpandingPanel
      collapseProps={{ id: 'w-varslerPanel-id' }}
      className='w-varslerPanel s-border'
      open={!widget.options.collapsed}
      highContrast={highContrast}
      onClick={onExpandablePanelChange}
      heading={<Systemtittel className='pb-3'>{t('ui:widget-overview-notifications')}</Systemtittel>}
    >
      <div className='row'>
        <div className='col-md-4'>
          <div style={{ display: 'none' }}>
            <div><label className='skjemaelement__label d-inline-block'>{t('pinfo:sb-sakId')}</label>: {sakId}</div>
            <div><label className='skjemaelement__label d-inline-block'>{t('ui:aktoerId')}</label>: {aktoerId}</div>
            <div><label className='skjemaelement__label d-inline-block'>{t('pinfo:sb-sakType')}</label>: {sakType}</div>
          </div>
          <Veileder
            tekst={(
              <>
                {!_.isEmpty(invite)
                  ? <div dangerouslySetInnerHTML={{ __html: t(invite!.message) }} />
                  : <div dangerouslySetInnerHTML={{ __html: t('ui:widget-overview-sendNotification-description', { user: user }) }} />}
              </>
            )}
            type={invite ? (invite.status === 'ERROR' ? 'feilmelding' : 'suksess') : 'normal'}
            fargetema={invite ? (invite.status === 'ERROR' ? 'feilmelding' : 'suksess') : 'normal'}
            posisjon='bunn'
          >
            <VeilederSVG />
          </Veileder>
          <div className='text-center'>
            <VerticalSeparatorDiv />
            <HighContrastHovedknapp
              data-testid='w-varslerPanel__invite-button-id'
              disabled={isInvitingPinfo}
              spinner={isInvitingPinfo}
              onClick={onInviteButtonClick}
            >
              {isInvitingPinfo ? t('ui:sending') : t('ui:widget-overview-sendNotification-button')}
            </HighContrastHovedknapp>
          </div>
        </div>
        <div className='col-md-8'>
          <Title>
            <div />
            <RefreshButton
              className='w-varslerPanel__refresh-button'
              labelRefresh={t('ui:refresh')}
              rotating={!isReady}
              onRefreshClicked={onRefreshHandle}
            />
          </Title>
          {!_.isEmpty(items) ? (
            <Table
              items={items}
              searchable
              selectable={false}
              sortable
              loading={!isReady}
              columns={[
                {
                  id: 'type',
                  label: t('ui:sentDocuments'),
                  type: 'object',
                  needle: /* istanbul ignore next */ (it: Item) => it.toLowerCase(),
                  renderCell: (item: any) => (
                    <FlexDiv>
                      <LineMessageSent />
                      <HorizontalSeparatorDiv />
                      <label>{item.type}</label>
                    </FlexDiv>
                  )
                },
                { id: 'sender', label: t('ui:sender'), type: 'string' },
                { id: 'date', label: t('ui:dateandtime'), type: 'date' }
              ]}
            />
          ) : (
            <NoMessage>
              {t('ui:widget-overview-sendNotification-noMessages')}
            </NoMessage>
          )}
        </div>
      </div>
    </ExpandingPanel>
  )
}

VarslerPanel.propTypes = {
  initialFiles: PT.objectOf(VarslerPropType.isRequired),
  onUpdate: PT.func.isRequired,
  widget: PT.any.isRequired // anyWidgetPropType.isRequired
}

export default VarslerPanel
