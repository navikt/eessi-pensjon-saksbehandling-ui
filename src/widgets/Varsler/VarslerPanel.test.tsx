import { sendInvite } from 'actions/pinfo'
import { getStorageFile, listStorageFiles } from 'actions/storage'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { stageSelector } from 'setupTests'
import { VarslerPanel, VarslerPanelProps, VarslerPanelSelector } from './VarslerPanel'

jest.mock('actions/pinfo', () => ({
  sendInvite: jest.fn()
}))

jest.mock('actions/storage', () => ({
  getStorageFile: jest.fn(),
  listStorageFiles: jest.fn()
}))

describe('widgets/Varsler/VarslerPanel', () => {
  let wrapper: ReactWrapper
  const defaultSelector: VarslerPanelSelector = {
    aktoerId: '10293847565',
    file: undefined,
    fileList: undefined,
    isInvitingPinfo: false,
    invite: undefined,
    person: {},
    sakId: '123',
    sakType: 'Alderspensjon'
  }

  const initialMockProps: VarslerPanelProps = {
    onUpdate: jest.fn(),
    widget: {
      i: 'i',
      type: 'varsler',
      title: 'Varsler',
      visible: true,
      options: {
        collapsed: false
      }
    }
  }

  beforeEach(() => {
    stageSelector(defaultSelector, {})
    wrapper = mount(<VarslerPanel {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.w-varslerPanel')).toBeTruthy()
  })

  it('With no params', () => {
    stageSelector(defaultSelector, { sakId: undefined })
    wrapper = mount(<VarslerPanel {...initialMockProps} />)
    expect(wrapper.exists('.w-varslerPanel__noParams-title')).toBeTruthy()
    expect(wrapper.exists('Veileder')).toBeTruthy()
  })

  it('UseEffect: There is a file list', () => {
    (getStorageFile as jest.Mock).mockReset()
    stageSelector(defaultSelector, ({ fileList: ['2020-01-01T11:11:11.json'] }))
    wrapper = mount(<VarslerPanel {...initialMockProps} />)
    expect(getStorageFile).toHaveBeenCalledWith({
      userId: defaultSelector.aktoerId,
      namespace: 'varsler',
      file: defaultSelector.sakId + '___2020-01-01T11:11:11.json'
    }, {
      notification: false
    })
  })

  it('UseEffect: There is a file list and a file to add', () => {
    (listStorageFiles as jest.Mock).mockReset();
    (getStorageFile as jest.Mock).mockReset()
    stageSelector(defaultSelector, ({
      fileList: ['2020-02-02T12:22:22.json', '2020-01-01T11:11:11.json', '2020-03-03T13:33:33.json'],
      file: {
        tittel: 'Mock File Name 3',
        fulltnavn: 'Sender 3',
        timestamp: '2020-01-03T13:33:33'
      }
    }))
    const mockInitialFiles = {
      '2020-02-02T12:22:22.json': {
        tittel: 'Mock File Name 1',
        fulltnavn: 'Sender 1',
        timestamp: '2020-01-02T12:22:22'
      },
      '2020-01-01T11:11:11.json': {
        tittel: 'Mock File Name 2',
        fulltnavn: 'Sender 2',
        timestamp: '2020-01-01T11:11:11'
      }
    }
    wrapper = mount(<VarslerPanel {...initialMockProps} initialFiles={mockInitialFiles} />)
    expect(getStorageFile).toHaveBeenCalledWith({
      userId: defaultSelector.aktoerId,
      namespace: 'varsler',
      file: defaultSelector.sakId + '___2020-03-03T13:33:33.json'
    }, {
      notification: false
    })
  })

  it('Refresh button triggers refresh action', () => {
    (listStorageFiles as jest.Mock).mockReset()
    wrapper.find('.w-varslerPanel__refresh-button a').hostNodes().simulate('click')
    expect(listStorageFiles).toHaveBeenCalledWith({
      userId: defaultSelector.aktoerId,
      namespace: 'varsler___' + defaultSelector.sakId
    })
  })

  it('Invite button triggers a invite action', () => {
    (sendInvite as jest.Mock).mockReset()
    wrapper.find('.w-varslerPanel__invite-button').hostNodes().simulate('click')
    expect(sendInvite).toBeCalledWith({
      aktoerId: defaultSelector.aktoerId,
      sakId: defaultSelector.sakId
    })
  })

  it('Expandable panel triggers widget update', () => {
    (initialMockProps.onUpdate as jest.Mock).mockReset()
    wrapper.find('.ekspanderbartPanel .ekspanderbartPanel__knapp').hostNodes().simulate('click')
    expect(initialMockProps.onUpdate).toBeCalledWith(expect.objectContaining({
      options: {
        collapsed: !initialMockProps.widget.options.collapsed
      }
    }))
  })
})
