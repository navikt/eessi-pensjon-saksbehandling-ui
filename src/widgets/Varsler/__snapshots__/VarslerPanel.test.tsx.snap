// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`widgets/Varsler/VarslerPanel Renders 1`] = `
<VarslerPanel
  onUpdate={[MockFunction]}
  widget={
    Object {
      "i": "i",
      "options": Object {
        "collapsed": false,
      },
      "title": "Varsler",
      "type": "varsler",
      "visible": true,
    }
  }
>
  <ow
    className="w-varslerPanel s-border"
    collapseProps={
      Object {
        "id": "w-varslerPanel-id",
      }
    }
    heading={
      <Systemtittel
        className="pb-3"
      >
        ui:widget-overview-notifications
      </Systemtittel>
    }
    onClick={[Function]}
    open={true}
  >
    <div
      className="c-expandingpanel ekspanderbartPanel w-varslerPanel s-border ekspanderbartPanel--apen"
      style={Object {}}
    >
      <div
        aria-controls="w-varslerPanel-id"
        aria-expanded={true}
        className="ekspanderbartPanel__hode"
        onClick={[Function]}
      >
        <div
          className="ekspanderbartPanel__flex-wrapper"
        >
          <Systemtittel
            className="pb-3"
          >
            <n
              className="pb-3"
              type="systemtittel"
            >
              <h2
                className="typo-systemtittel pb-3"
              >
                ui:widget-overview-notifications
              </h2>
            </n>
          </Systemtittel>
          <button
            aria-expanded={true}
            aria-label="open"
            className="ekspanderbartPanel__knapp"
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <span
              className="ekspanderbartPanel__indikator"
            />
          </button>
        </div>
      </div>
      <t
        id="w-varslerPanel-id"
        isOpened={true}
        onRest={[Function]}
      >
        <n
          checkTimeout={50}
          id="w-varslerPanel-id"
          initialStyle={
            Object {
              "height": "auto",
              "overflow": "initial",
            }
          }
          isOpened={true}
          onRest={[Function]}
          onWork={[Function]}
          theme={
            Object {
              "collapse": "ReactCollapse--collapse",
              "content": "ReactCollapse--content",
            }
          }
        >
          <div
            className="ReactCollapse--collapse"
            style={
              Object {
                "height": "auto",
                "overflow": "initial",
              }
            }
          >
            <div
              className="ReactCollapse--content"
            >
              <article
                className="ekspanderbartPanel__innhold"
              >
                <Row>
                  <div
                    className="row"
                  >
                    <div
                      className="col-md-4"
                    >
                      <div
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                      >
                        <div>
                          <label
                            className="skjemaelement__label d-inline-block"
                          >
                            pinfo:sb-sakId
                          </label>
                          : 
                          123
                        </div>
                        <div>
                          <label
                            className="skjemaelement__label d-inline-block"
                          >
                            ui:aktoerId
                          </label>
                          : 
                          10293847565
                        </div>
                        <div>
                          <label
                            className="skjemaelement__label d-inline-block"
                          >
                            pinfo:sb-sakType
                          </label>
                          : 
                          Alderspensjon
                        </div>
                      </div>
                      <Veileder
                        fargetema="normal"
                        posisjon="bunn"
                        storrelse="M"
                        tekst={
                          <React.Fragment>
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "ui:widget-overview-sendNotification-description",
                                }
                              }
                            />
                          </React.Fragment>
                        }
                        type="normal"
                      >
                        <div
                          className="nav-veileder nav-veileder--bunn"
                        >
                          <div
                            className="nav-veileder__frame nav-veileder__frame--m"
                          >
                            <ForwardRef(SvgNavPensjonVeileder)>
                              <svg>
                                NavPensjonVeileder.svg
                              </svg>
                            </ForwardRef(SvgNavPensjonVeileder)>
                          </div>
                          <i
                            className="nav-veileder__snakkeboblePil nav-veileder__snakkeboblePil--bunn"
                          />
                          <div
                            className="nav-veileder__snakkeboble nav-veileder__snakkeboble--bunn"
                          >
                             
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "ui:widget-overview-sendNotification-description",
                                }
                              }
                            />
                          </div>
                        </div>
                      </Veileder>
                      <div
                        className="text-center"
                      >
                        <Hovedknapp
                          className="w-varslerPanel__invite-button mt-3"
                          disabled={false}
                          id="w-varslerPanel__invite-button-id"
                          onClick={[Function]}
                          spinner={false}
                        >
                          <KnappBase
                            autoDisableVedSpinner={false}
                            className="w-varslerPanel__invite-button mt-3"
                            disabled={false}
                            htmlType="submit"
                            id="w-varslerPanel__invite-button-id"
                            inaktivKlasseVedDisabled={false}
                            kompakt={false}
                            mini={false}
                            onClick={[Function]}
                            spinner={false}
                            type="hoved"
                          >
                            <button
                              className="knapp w-varslerPanel__invite-button mt-3 knapp--hoved"
                              disabled={false}
                              id="w-varslerPanel__invite-button-id"
                              onClick={[Function]}
                              type="submit"
                            >
                              ui:widget-overview-sendNotification-button
                            </button>
                          </KnappBase>
                        </Hovedknapp>
                      </div>
                    </div>
                    <div
                      className="col-md-8"
                    >
                      <div
                        className="w-varslerPanel__title"
                      >
                        <div />
                        <RefreshButton
                          className="w-varslerPanel__refresh-button"
                          labelRefresh="ui:refresh"
                          onRefreshClicked={[Function]}
                          rotating={true}
                        >
                          <div
                            className="c-refreshbutton w-varslerPanel__refresh-button"
                          >
                            <a
                              className="lenke rotating"
                              href="#refresh"
                              onClick={[Function]}
                              title="ui:refresh"
                            >
                              <Icons
                                kind="refresh"
                              >
                                <FontAwesomeIcon
                                  border={false}
                                  className=""
                                  fixedWidth={false}
                                  flip={null}
                                  icon={
                                    Object {
                                      "icon": Array [
                                        512,
                                        512,
                                        Array [],
                                        "f2f1",
                                        "M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z",
                                      ],
                                      "iconName": "sync-alt",
                                      "prefix": "fas",
                                    }
                                  }
                                  inverse={false}
                                  kind="refresh"
                                  listItem={false}
                                  mask={null}
                                  pull={null}
                                  pulse={false}
                                  rotation={null}
                                  size="1x"
                                  spin={false}
                                  swapOpacity={false}
                                  symbol={false}
                                  title=""
                                  transform={null}
                                >
                                  <svg
                                    aria-hidden="true"
                                    className="svg-inline--fa fa-sync-alt fa-w-16 fa-1x "
                                    data-icon="sync-alt"
                                    data-prefix="fas"
                                    focusable="false"
                                    kind="refresh"
                                    role="img"
                                    style={Object {}}
                                    viewBox="0 0 512 512"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z"
                                      fill="currentColor"
                                      style={Object {}}
                                    />
                                  </svg>
                                </FontAwesomeIcon>
                              </Icons>
                            </a>
                          </div>
                        </RefreshButton>
                      </div>
                      <div
                        className="p-2 w-varslerPanel__noMessages font-italic"
                      >
                        ui:widget-overview-sendNotification-noMessages
                      </div>
                    </div>
                  </div>
                </Row>
              </article>
            </div>
          </div>
        </n>
      </t>
    </div>
  </ow>
</VarslerPanel>
`;
